### PR TITLE
Modify `BigArrays` to take name of circuit breaker

### LINKED ISSUE #31435: BigArrays#newByteArray always uses REQUEST breaker
When you create a byte array using BigArrays, it always increases the
`REQUEST` breaker. The `REQUEST` is described as:

```
/**
 *  The request breaker tracks memory used for particular requests. This
 *  includes allocations for things like the cardinality aggregation, and
 *  accounting for the number of buckets used in an aggregation request.
 *  Generally the amounts added to this breaker are released after a request
 *  is finished.
 */
```

However, there is also a `IN_FLIGHT_REQUESTS` breaker. This breaker is
described as:

```
/**
 * The in-flight request breaker tracks bytes allocated for reading and
 * writing requests on the network layer.
 */
```

When we serialize a message for writing to a channel in the http or tcp
transports, we use `BigArrays` which manipulates the `REQUEST` breaker.
However, based on the descriptions, it seems like this should be the
`IN_FLIGHT_REQUESTS` breaker.

### PR DESCRIPTION
**Pull request by Tim-Brooks**
This commit modifies `BigArrays` to take a circuit breaker name and
the circuit breaking service. The default instance of `BigArrays` that
is passed around everywhere always uses the request breaker. At the
network level, we want to be using the inflight request breaker. So this
change will allow that.

Additionally, as this change moves away from a single instance of
`BigArrays`, the class is modified to not be a `Releasable` anymore.
Releasing big arrays was always dispatching to the `PageCacheRecycler`,
so this change makes the `PageCacheRecycler` the class that needs to be
managed and torn-down.

Finally, this commit closes #31435 be making the serialization of
transport messages use the inflight request breaker. With this change,
we no longer push the global `BigArrays` instnace to the network level.

### CODE REVIEW

**Comment by tbrooks8** [comment]
I modified this to reflect the fact that you must always access the circuit breaker service due to potential settings changes.

File: server/src/main/java/org/elasticsearch/common/util/BigArrays.java [hunk 1]
```diff
@@ -380,9 +376,15 @@ public BigArrays(PageCacheRecycler recycler, @Nullable final CircuitBreakerServi
 
     // public for tests
     public BigArrays(PageCacheRecycler recycler, @Nullable final CircuitBreakerService breakerService, boolean checkBreaker) {
```
Comment (jpountz) [inline server/src/main/java/org/elasticsearch/common/util/BigArrays.java]: can we remove this constructor and make call sites pass `CircuitBreaker.REQUEST` as a breaker name?

**Review by jpountz — APPROVED** [review]
Thanks @tbrooks8, this looks much cleaner to me this way! I'd just like that we keep a single BigArrays constructor.

**Comment by tbrooks8** [comment]
@jpountz - Thanks. 

There is now a public ctor that takes recycler, breaker service and name. And a protected ctor that takes whether we should check breaker (exists for `MockBigArrays`).

**Review by jpountz — APPROVED** [review]
LGTM

**Pull request merged**